src/Components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,51 @@
+import { Autocomplete, FormControl, FormHelperText, TextField } from '@mui/material';
+import { useMemo } from 'react';
+import { FormattedMessageType } from '../../Types/Questions';
+
+type Props = {
+  value: string | null;
+  setValue: (newValue: string | null) => void;
+  options: string[];
+  label: FormattedMessageType;
+  showError: boolean;
+  errorMessage: FormattedMessageType;
+};
+
+export default function AutoComplete({ options, label, value, setValue, showError, errorMessage }: Props) {
```

Comment:
What do you think about using react-hook-form and zod for these new components?

Reply:
I would have to refactor the whole household data block question.